schedule/src/main/java/com/droidcon/schedule/data/repository/SessionsRepository.kt

```diff
@@ -1,20 +1,29 @@
 package com.droidcon.schedule.data.repository
 
 import com.droidcon.schedule.data.SessionData
+import com.droidcon.schedule.data.SessionsCacheStorage
 import com.droidcon.schedule.data.network.SessionsApiClient
 import com.droidcon.schedule.data.network.toSessionData
 import javax.inject.Inject
 
 
 class SessionsRepository @Inject constructor(
+    private val sessionsCacheStorage: SessionsCacheStorage,
     private val sessionsApiClient: SessionsApiClient
 ) {
 
     suspend fun getAllSessions(): List<SessionData> =
-        sessionsApiClient.getSessions()
-            .flatMap { sessionsGroup ->
-                sessionsGroup.sessions.map { networkSession ->
-                    networkSession.toSessionData(sessionsGroup.groupName)
-                }
-            }
+        if (sessionsCacheStorage.get().isNullOrEmpty()) {
```

Comment:
We could avoid the not null operator with the next:
```
val cachedSessions = sessionsCacheStorage.get()
return if(cachedSessions.isNullOrEmpty()) {
    getAllSessionsFromApi()
} else {
    cachedSessions
}
```

Reply:
Cool thanks for the feedback. I'm chaging that code now that I'm adiding persistence with room. But it will be in that format:
`    suspend fun getAllSessions(): List<SessionData> {
        val sessions = getAllSessionsFromDisk()
        return if (sessions.isNullOrEmpty()) {
            getAllSessionsFromApi()
        } else {
            sessions
        }
    }`